### ISSUE TITLE
[PE-D][Tester D] Tags are case-sensetive, yet `find` finds them in case-insensetive manner

### ISSUE THREAD

**Issue by nus-se-script** [AY2425S1-CS2103T-T10-3/tp#195]
Reproduction:
1. `add n/Jing Wen i/E0123456 p/12345678 e/jw@jw.com t/y2 t/Y3`
2. `add n/test1 i/E0123457 p/12345671 e/jwa@jw.com t/y1`
3. `add n/test2 i/E0123458 p/12345672 e/jwb@jw.com t/Y4`
4. `find t/y`
5. Everything gets returned

When adding entries with tags that differ only by case (e.g., y2 vs Y2), the tags are treated as distinct. However, when using the find command to search by tags, the search is performed in a case-insensitive manner, which causes unintended matches. As a result, tags that are case-sensitive are inconsistently handled during searches, returning all records that contain tags with similar case-insensitive matches.

This bug can cause confusion, especially for users relying on strict case-sensitive tagging for categorization. It becomes impossible to search specifically for entries tagged with lowercase vs. uppercase tags or mixed-case tags. 

It also would be sensible to define tags as case-insensetive, but this should be made consistent across all commands (i.e. `add` treats it as case-sensitive`).

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: shotnothing/ped#13

**Comment by Domokunx** [comment]
I think the case-sensitive finding is a valid concern.

But in the context of NUS TAs, I don't think the case sensitivity of tags makes a big impact?

Whats the difference between 'y2' VS 'Y2'? If we do find `Y` and `y2` is not shown, then would this also be a feature flaw?

I think its subjective, either ignore or planned enhancement to enable case sensitive `find`

**Comment by yockcheng** [comment]
I don't think this is a valid flaw. I can't think of any scenario where 2 identical tags with different letter casing have differing meanings. Thus, there is no reason for the user to find using a tag with some specific letter casing since the the tags would have the same meaning regardless if its upper or lower case.

**Comment by sayglenn** [comment]
Again, critiquing an implementation rather than its correctness. Reject and will close.
